File: src/main/cpp/addon/AddonConfig.cpp
#include "include/AddonConfig.h"

namespace ravex {
namespace addon {

void AddonConfig::set(const std::string& key, const std::string& val) {
    settings[key] = val;
}

std::string AddonConfig::get(const std::string& key, const std::string& def) {
    auto it = settings.find(key);
    return it != settings.end() ? it->second : def;
}

}
}

File: src/main/java/ravex/addon/util/AddonConfig.java
package ravex.addon.util;

import java.io.File;
import java.util.HashMap;
import java.util.Map;

public class AddonConfig {
    private final Map<String, String> properties = new HashMap<>();
    private final File file;

    public AddonConfig(File file) {
        this.file = file;
    }

    public void set(String key, String value) { properties.put(key, value); }
    public String get(String key, String def) { return properties.getOrDefault(key, def); }
    public void load() {}
    public void save() {}
}
